include/openbabel/tautomer.h

```diff
@@ -59,31 +61,74 @@ namespace OpenBabel {
    * WriteSmilesFunctor functor;
    * // write all tautomers to std::cout
    * EnumerateTautomers(mol, functor);
-   * // write the original smiles if there are no tautomers found
-   * if (!functor.foundTautomers) {
-   *   OBConversion conv;
-   *   conv.SetOutFormat("can");
-   *   std::cout << conv.WriteString(&mol);
-   * }
    * @endcode
-   */  
+   */
   class OBAPI TautomerFunctor
   {
     public:
       virtual ~TautomerFunctor() {}
       /**
        * This function is called every time a tautomer is discovered.
+       *
+       * @param mol The tautomer.
        */
       virtual void operator()(OBMol *mol) = 0;
-  }; 
+  };
+
+  /**
+   * Functor for tautomer enumeration algorithm. Subclass this class to write
+   * your own functor. This functor will be called for every unique tautomer
+   * that is found.
+   *
+   * Example to write all unique tautomer smiles to std::cout:
+   * @code
+   * class WriteSmilesFunctor : public UniqueTautomerFunctor
+   * {
+   *   public:
+   *     void operator()(OBMol *mol, const std::string &smiles)
+   *     {
+   *       std::cout << smiles << std::endl;
+   *     }
+   * };
+   * @endcode
+   *
+   * This functor can be used in the following way:
+   * @code
+   * OBMol mol;
+   * ...
+   * WriteSmilesFunctor functor;
+   * // write all unique tautomers to std::cout
+   * EnumerateTautomers(mol, functor);
+   * @endcode
+   */
+  class OBAPI UniqueTautomerFunctor : public TautomerFunctor
```

Comment:
I'll mark this for 3.1 when we can add new API

Reply:
Should I make a separate version with only the bug fixes and no API changes for 3.0.1?

Reply:
If you think you can provide only the bug fixes with no API changes, that would be great. There's clearly interest in a 3.0.1 release.